webpack/components/extensions/Hosts/BulkActions/BulkChangeHostCollectionsModal/BulkChangeHostCollectionsModal.js

```diff
@@ -0,0 +1,323 @@
+  const replacementResponse = isOpen ? false : { response: {} };
+
+  const hostCollectionsResponse = useTableIndexAPIResponse({
+    replacementResponse, // don't fetch data if modal is closed
+    apiUrl: hostCollectionsUrlForOrg,
+    apiOptions,
+    defaultParams: { per_page: DEFAULT_PER_PAGE },
+  });
+
+  const {
+    response: {
+      results: hostCollectionsResults,
+      ...hostCollectionsMetadata
+    },
+    setAPIOptions,
+    status: hostCollectionsLoadingStatus,
+  } = hostCollectionsResponse;
+
+  const {
+    total, page, subtotal, per_page: perPage,
+  } = hostCollectionsMetadata;
+
+  const hostCollectionsBulkSelect = useBulkSelect({
+    results: hostCollectionsResults,
+    metadata: { total, page, selectable: subtotal },
+    idColumn: 'id',
+  });
+
+  const {
+    setParamsAndAPI: setHostCollectionsParamsAndAPI,
+  } = useSetParamsAndApiAndSearch({
+    defaultParams: { search: '' },
+    apiOptions,
+    setAPIOptions,
+    updateSearchQuery: hostCollectionsBulkSelect.updateSearchQuery,
+    pushToHistory: false,
+  });
+
+  const {
+    selectPage,
+    selectNone,
+    selectOne,
+    isSelected,
+    selectedCount,
+    selectedResults,
+    areAllRowsSelected,
+    areAllRowsOnPageSelected,
+    updateSearchQuery,
+  } = hostCollectionsBulkSelect;
+
+  const pageStats = getPageStats({ total: subtotal, page, perPage });
+
+  // Update HCIds when selections change
+  useEffect(() => {
+    if (isOpen) {
+      const selectedIds = selectedResults.map(result => result.id);
+      setHCIds(selectedIds);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedCount, isOpen]);
+
+  const selectionToolbar = (
+    <ToolbarItem key="selectAll">
+      <SelectAllCheckbox
+        {...{
+          selectAll: noop,
+          selectOne,
+          updateSearchQuery,
+          selectNone,
+          selectPage,
+          selectedCount,
+          pageRowCount: pageStats.pageRowCount,
+          areAllRowsSelected,
+          areAllRowsOnPageSelected,
+        }}
+        totalCount={total}
+        areAllRowsOnPageSelected={areAllRowsOnPageSelected()}
+        areAllRowsSelected={areAllRowsSelected()}
+      />
+    </ToolbarItem>
+  );
+
+  const handleModalClose = () => {
+    setHCIds([]);
+    setAddRadioChecked(true);
+    selectNone();
+    closeModal();
+  };
+
+  const handleError = (error) => {
+    const message = error?.response?.data?.error?.message || __('An error occurred while updating host collections');
+    dispatch(addToast({ type: 'danger', message }));
+  };
+
+  const handleSuccess = (response) => {
+    const messages = response?.data?.displayMessages;
+    const message = Array.isArray(messages) && messages.length > 0
+      ? (
+        <>
+          {messages.map((msg, index) => (
+            <div key={`host-status-${index}`}>{msg}</div>
+          ))}
+        </>
+      )
+      : __('Host collections updated successfully');
```

Comment:
We don't use the word "successfully" in UI messages. It's redundant.

```suggestion
      : __('Host collections updated');
```